tests/app/forms/field_handlers/test_select_handler.py

```diff
@@ -54,3 +56,41 @@ class TestForm(Form):
     assert form.test_field.label.text == radio_json["label"]
     assert form.test_field.description == radio_json["guidance"]
     assert form.test_field.choices == expected_choices
+
+
+def test_get_field_with_bad_choices(value_source_resolver):
+    radio_json = {
+        "guidance": "",
+        "id": "choose-your-side-answer",
+        "label": "Choose a side",
+        "mandatory": True,
+        "options": [
+            {
+                "label": "Light Side",
+                "value": "Light Side",
+                "description": "The light side of the Force",
+            },
+            {
+                "label": "Dark Side",
+                "value": "Dark Side",
+                "description": "The dark side of the Force",
+            },
+            {"label": "I prefer Star Trek", "value": "I prefer Star Trek"},
+            {"label": "Other", "value": "Other"},
+        ],
+        "q_code": "20",
+        "type": "Radio",
+        "validation": {"messages": {"MANDATORY_RADIO": "This answer is required"}},
+    }
+    with patch.object(
+        SelectHandler, "build_choices_with_detail_answer_ids", return_value=None
+    ):
+        handler = SelectHandler(radio_json, value_source_resolver, error_messages)
+
+        class TestForm(Form):
+            test_field = handler.get_field()
+
+        form = TestForm()
+
+        with pytest.raises(TypeError):
```

Comment:
I was under the impression from the code above that TypeError was raised on empty choices and this should be ValidationError unless I have missed something
```        
if not self.choices:
      raise TypeError(self.gettext("Not a valid choice"))
...
 else:
      raise ValidationError(self.gettext("Not a valid choice"))

````

Reply:
Yeah the TypeError was actually being raised from trying to iterate over a `None` object which was from a dodgy mock which is why the test was passing. Turns out I actually need to test presence of `form.errors` instead.